pkg/devfile/parser/context/content.go

```diff
@@ -65,6 +65,19 @@ func (d *DevfileCtx) SetDevfileContent() error {
 	return nil
 }
 
+// SetDevfileContentFromBytes reads devfile and if devfile is in YAML format converts it to JSON
+func (d *DevfileCtx) SetDevfileContentFromBytes(data []byte) error {
```

Comment:
This function should be called from SetDevfileContent once bytes are extracted from file to avoid code duplication

Reply:
Done